restic/restore.go

```diff
@@ -108,8 +108,10 @@ func (r *RestoreStruct) restoreCommand(snapshotID, method string, snaps []Snapsh
 		fmt.Printf("Snapshot %v is being restored.\n", snapshot.Time)
 	} else {
 		for i := range snaps {
-			if snaps[i].ID == snapshotID {
+			// Doing substrings so we can also use short IDs here.
+			if snaps[i].ID[0:len(snapshotID)] == snapshotID {
 				snapshot = snaps[i]
+				break
```

Comment:
Interesting; previously the code used to take the last match, now it takes the first; dunno if that's a problem, but I'm just making an observation.

Reply:
Each snapshot ID should be unique, so after finding a match there's really no point to loop through the others. It's a minor speedup.